### PR TITLE
Improve Ruby Examples for email validation REST API calls

### PR DESCRIPTION
**Pull request by honzasterba**
- use RestClient's GET built-in GET parameter serialisation
- use single line hash declaration

### CODE REVIEW

File: source/samples/get-parse.rst [hunk 1]
```diff
@@ -51,12 +51,12 @@
...
-   url_params[:addresses] = "Alice <alice@example.com>,bob@example.com"
-   query_string = url_params.collect {|k, v| "#{k.to_s}=#{CGI::escape(v.to_s)}"}.
-     join("&")
-   RestClient.get "https://api:pubkey-5ogiflzbnjrljiky49qxsiozqef5jxp7"\
-   "@api.mailgun.net/v3/address/parse?#{query_string}"
+   url_params = { addresses:  "Alice <alice@example.com>,bob@example.com" }
+   public_key = "pubkey-5ogiflzbnjrljiky49qxsiozqef5jxp7"
+   parse_url = "https://api:#{public_key}@api.mailgun.net/v3/address/parse"
+   RestClient::Request.execute method: :get, url: parse_url,
+                                       headers: { params: url_params },
```
Comment (thrawn01) [inline source/samples/get-parse.rst]: why pass in the url_params via headers?

Comment (honzasterba) [reply source/samples/get-parse.rst]: Its counter-intuitive, but when RestClient sees a header named params it removes it and coverts to query-string params for GET requests. Does away with the whole manual escaping which is ugly

**Comment by honzasterba** [comment]
The not about Request.execute is subjective, the shorthand get/post methods do not support any additional args appart from body/headers/query so to do for example basic-auth you need to do execute or construct ugly URLs that then RestClient parses out anyways

**Comment by thrawn01** [comment]
Thanks for explaining, This just makes me glad I don't write ruby anymore.

**Pull request merged**